lib/archethic/transaction_chain/transaction/validation_stamp/ledger_operations.ex

```diff
@@ -288,15 +288,25 @@ defmodule Archethic.TransactionChain.Transaction.ValidationStamp.LedgerOperation
         contract_context \\ nil
       ) do
     # Since AEIP-19 we can consume from minted tokens
-    inputs = inputs ++ tokens_to_mint
+    consolidated_inputs = tokens_to_mint ++ inputs
 
-    %{uco: uco_balance, token: tokens_balance} = ledger_balances(inputs)
+    %{uco: uco_balance, token: tokens_balance} = ledger_balances(consolidated_inputs)
     %{uco: uco_to_spend, token: tokens_to_spend} = total_to_spend(fee, movements)
 
     if sufficient_funds?(uco_balance, uco_to_spend, tokens_balance, tokens_to_spend) do
       consumed_utxos =
-        get_inputs_to_consume(
-          inputs,
+        consolidated_inputs
+        |> Enum.map(fn
+          utxo = %UnspentOutput{from: ^change_address} ->
```

Comment:
I think this burn address modification should be done directly when we create the utxo at `LedgerOperations.get_utxos_from_transaction`

Reply:
Not completely because we need them as unspent output with the change address after , if none are spent.